### PR TITLE
[693] Add the ability to change the views of the workbench

### PR DESCRIPTION
**Pull request by pcdavid**
- [693] Contribute the ADR
- [693] Use the same type for props of all existing views
- [WIP 693] Make the workbench panels configurable

### Type of this PR 

- [ ] Bug fix
- [x] New feature or improvement
- [ ] Documentation
- [ ] Cleanup
- [ ] Test
- [ ] Build/Releng

### Issue(s)

...

### What does this PR do?

...

### Screenshot/screencast of this PR

...
 

### Potential side effects

...

### How to test this PR?

- [ ] Frontend Unit tests
- [ ] Backend Unit tests
- [ ] Cypress : please specify
- [ ] Manual Test : please specify

### Checklist

- [ ] I have read CONTRIBUTING carefully.
- [ ] I have signed the [Contributor License Agreement](http://www.eclipse.org/legal/CLA.php).
- [ ] All my commits are signed-off (`-s`) with my mail address of my Eclipse Account.
- [ ] I have covered my changes by unit tests or integration tests or manual tests.
- [ ] All tests pass.
- [ ] I have updated the documentation accordingly
- [ ] New React components are availables in storybook

### CODE REVIEW

File: doc/adrs/046_make_the_workbench_panels_content_configurable.adoc [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+<Workbench
+  editingContextId={project.currentEditingContext.id}
+  initialRepresentationSelected={representation}
+  onRepresentationSelected={onRepresentationSelected}
+  readOnly={project.accessLevel === 'READ'}>
+  <WorkbenchViewContribution side="left" title="Validation" component={ValidationWebSocketContainer} />
+  <WorkbenchViewContribution side="left" title="Explorer" component={ExplorerWebSocketContainer} />
+  <WorkbenchViewContribution side="right" title="Details" component={PropertiesWebSocketContainer} />
+  <WorkbenchViewContribution side="right" title="Representations" component={RepresentationsWebSocketContainer} />
+  <TreeItemContextMenuContribution canHandle={(item: TreeItemType) => item.kind.startsWith('obeoStudio://document')}
```
Comment (sbegaudeau) [inline doc/adrs/046_make_the_workbench_panels_content_configurable.adoc:25]: `siriusWeb://` instead of `obeoStudio://`

**Review by sbegaudeau — COMMENTED** [review]
Technically, removing the interfaces of the existing views like `PropertiesWebSocketContainerProps` is an API break. I suspect that some other projects were using those interfaces. We should at least mention that they should switch to the newly introduced `WorkbenchViewComponentProps`

File: frontend/src/workbench/Site.tsx [hunk 2]
```diff
@@ -98,63 +96,48 @@ const CustomCollapse = (props: CollapseProps) => {
   );
 };
 
-const DETAILS_PANEL_NAME = 'details';
-const REPRESENTATIONS_PANEL_NAME = 'representations';
-
-export const RightSite = ({ editingContextId, selection, setSelection, readOnly }: RightSiteProps) => {
+export const Site = ({ editingContextId, selection, setSelection, readOnly, contributions }: SiteProps) => {
```
Comment (sbegaudeau) [inline frontend/src/workbench/Site.tsx:99]: The code looks good, the only remaining things that should be improved is the ability to support more than two `WorkbenchView` per `Site`. We could probably use a vertical flex layout and say that the expanded accordion should take all the space available with `flex-grow: 1`

File: doc/adrs/046_make_the_workbench_panels_content_configurable.adoc (hunk 1 above)
Comment (pcdavid) [reply doc/adrs/046_make_the_workbench_panels_content_configurable.adoc]: Done

**Comment by pcdavid** [comment]
> Technically, removing the interfaces of the existing views like `PropertiesWebSocketContainerProps` is an API break. I suspect that some other projects were using those interfaces. We should at least mention that they should switch to the newly introduced `WorkbenchViewComponentProps`

Done

File: frontend/src/workbench/Site.tsx (hunk 2 above)
Comment (pcdavid) [reply frontend/src/workbench/Site.tsx:96]: Done

**Review by sbegaudeau — APPROVED** [review]
It looks nice 👍

I'll merge it after the build

**Comment by sbegaudeau** [comment]
I'll add a commit for the 2022.01.15 too

**Pull request merged**